sources/models/src/lib.rs

```diff
@@ -295,6 +295,7 @@ struct KubernetesSettings {
     shutdown_grace_period_for_critical_pods: KubernetesDurationValue,
     memory_manager_reserved_memory: HashMap<Identifier, KubernetesMemoryReservation>,
     memory_manager_policy: KubernetesMemoryManagerPolicy,
+    reserved_cpus: SingleLineString,
```

Comment:
I read the code of how the Kubernetes parses this line, and I think we should implement the same parsing logic in a struct. We already do something similar for other values that kubernetes expects in other places (see [KubernetesDurationValue]). This will prevent users from setting a random, since the API will refuse to apply the value, which provides a better experience than chasing errors in the kubelet logs. The new type could be `KubernetesCpuSet`, and implement what the kubernetes folks did [here].

[KubernetesDurationValue]: https://github.com/bottlerocket-os/bottlerocket/blob/d47e652b701c5b5e150146cb250971d263584f68/sources/models/modeled-types/src/kubernetes.rs#L820
[here]: https://github.com/kubernetes/utils/blob/d93618cff8a22d3aea7bf78d9d528fd859720c2d/cpuset/cpuset.go#L203

Reply:
Arguably this is not a kubernetes specific string - it's a kernel concept,  "cpuset string"  is effectively a standard.

https://docs.kernel.org/admin-guide/cgroup-v2.html#cpuset-interface-files

It is probably a useful concept to have in Bottlerocket, as my future PRs will include a lot more use of "cpuset string" that are not tied to Kubernetes.

So, if you want this, best it's `KernelCpuSet` - not `KubernetesCpuSet`

But it's almost academic - I lack the skills to do this in Rust ;-)

Reply:
https://github.com/bottlerocket-os/bottlerocket/commit/0c9d63c7b55fe8ce977b3cd54ca4c8ba21c550d2 I wrote up a quick approach to this to assist with getting you there. This should work for your needs but I didn't test the integration myself. I'm happy to help get you the rest of the way if needed on this change!

Reply:
negative case correctly gives.

```
[    4.753115] early-boot-config[1198]: Error PATCHing '/settings?tx=bottlerocket-launch': Status 400 when PATCHing /settings?tx=bottlerocket-launch: Json deserialize error: Unable to deserialize into KernelCpuSetValue: Invalid Kernel CpuSet value 'abc' at line 1 column 1698
```